mantle/cmd/kola/switchkernel.go

```diff
@@ -0,0 +1,249 @@
+				{
+					"filesystem": "root",
+					"path": "/var/home/core/switch-kernel.sh",
+					"contents": {
+						"source": "data:text/plain;base64,%s"
+					},
+					"mode": 110
+				}
+			]
+		}
+	}`, base64.StdEncoding.EncodeToString([]byte(switchKernelScript))))
+
+	flight, err := kola.NewFlight(kolaPlatform)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create new flight")
+	}
+	defer flight.Destroy()
+
+	c, err := flight.NewCluster(&platform.RuntimeConfig{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create new cluster")
+	}
+	defer c.Destroy()
+
+	m, err := c.NewMachine(userdata)
+	if err != nil {
+		return errors.Wrapf(err, "failed to spawn new machine")
+	}
+	defer m.Destroy()
+
+	err = testSwitchKernel(c)
+	if err != nil {
+		return errors.Wrapf(err, "failed switch kernel test")
+	}
+
+	return nil
+}
+
+// Drops Kernel RT RPM files under the directory `localPath` to `$homeDir/kernel-rt-rpms` directory in m
+func dropRpmFilesAll(m platform.Machine, localPath string) error {
+	fmt.Println("Dropping RT Kernel RPMs...")
+	re := regexp.MustCompile(`^kernel-rt-.*\.rpm$`)
+	files, err := ioutil.ReadDir(localPath)
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		filename := f.Name()
+		filepath := strings.TrimSuffix(localPath, "/") + "/" + filename
+		targetPath := fmt.Sprintf("%s/kernel-rt-rpms/%s", homeDir, filename)
+		if re.MatchString(filename) {
+			in, err := os.Open(filepath)
+			if err != nil {
+				return err
+			}
+			if err := platform.InstallFile(in, m, targetPath); err != nil {
+				return errors.Wrapf(err, "failed to upload %s", filename)
+			}
+		}
+	}
+	return nil
+}
+
+func switchDefaultToRtKernel(c platform.Cluster, m platform.Machine) error {
+	// run the script to switch from default kernel to rt kernel
+	fmt.Println("Switching from Default to RT Kernel...")
+	cmd := "sudo " + homeDir + "/switch-kernel.sh default rt-kernel " + homeDir + "/kernel-rt-rpms/"
+	stdout, stderr, err := m.SSH(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run %s", cmd)
+	}
+	fmt.Println(fmt.Sprintf("%s", stderr))
+	fmt.Println(fmt.Sprintf("%s", stdout))
+
+	// reboot the machine to switch kernel
+	fmt.Println("Rebooting machine...")
+	err = m.Reboot()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reboot machine")
+	}
+
+	// check if the kernel has switched to rt kernel
+	fmt.Println("Checking kernel type...")
+	cmd = "uname -v | grep -q 'PREEMPT RT'"
+	_, _, err = m.SSH(cmd)
+	if err == nil {
+		return errors.Wrapf(err, "failed to run %s", cmd)
+	}
+	fmt.Println("Switched to RT Kernel successfully!")
+
+	return nil
+}
+
+func switchRtKernelToDefault(c platform.Cluster, m platform.Machine) error {
+	// run the script to switch from rt kernel to default
+	fmt.Println("Switching from RT to Default Kernel...")
+	cmd := "sudo " + homeDir + "/switch-kernel.sh rt-kernel default"
+	stdout, stderr, err := m.SSH(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to run %", cmd)
```

Comment:
cmd/kola/switchkernel.go:206:10: Wrapf format % is missing verb at end of string

Reply:
You're missing the `s` of the `%s` in the format string.

Reply:
Yep I wanted to make a note I missed that